### PR TITLE
front login and register pages 

### PR DESCRIPTION
**Pull request by uniQueSense**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+  <Router>
+    <Route path="/" exact component={Login}/>
```
Comment (ParleJe) [inline src/App.js:10]: I think its redundant pathing. Path "/" will match "/login". Remove this like and try, if it works change it

File: src/Login.css [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+.login-php {
+    grid-row-start: span 2;
+    grid-column: 4 / span 2; 
+}
+
+.login-python {
+    grid-row: 4 / span 2;
+    grid-column-start: span 2;
+}
```
Comment (ParleJe) [inline src/Login.css:110]: These classes can be ID's

File: src/Login.css [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        width: 10em;
+        font-size: 1.5em;
+    }
+}
+
+.login-line hr {
+    width: 2em;
+    height: 0.5em;
+    border: none;
+    background: #F6F6F6;
```
Comment (ParleJe) [inline src/Login.css:201]: introduce css variable for colours. [link](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties)

File: src/Login.css [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+.login-place {
+    display: grid;
+    justify-items: center;
+    align-items: center;
+}
+
+.login-place input {
+    margin: 0.5em;
+    border: 0;;
```
Comment (ParleJe) [inline src/Login.css:166]: double semicolon

File: src/Login.css [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    }
+}
+
+@media (max-width: 350px) {
+    .login-place input {
+        width: 11.7em;
+        font-size: 1.5em;
+        padding: 0.1em 0.3em;
+    }
+}
```
Comment (ParleJe) [inline src/Login.css:188]: You don't change font-size, so why are you typing it again in another media query?

File: src/Login.css [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+@media (max-width: 300px) {
+    .login-place input {
+        width: 10em;
+        font-size: 1.5em;
```
Comment (ParleJe) [inline src/Login.css:193]: as above, why again?

File: src/Login.css [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+.login-button {
+    display: grid;
+    justify-items: center;
+    align-items: center;
+    border-radius: 1.5em;
+    height: 2.1em;
+    width: 17em;
+    font-size: 1.5em;
+    border-color: #F6F6F6;
```
Comment (ParleJe) [inline src/Login.css:220]: css variable

File: src/Login.css [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+    background: none;
+    color: #F6F6F6;
```
Comment (ParleJe) [inline src/Login.css:222]: css variable

File: src/Login.css [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+        font-size: 1.4em;
+    }
+}
+
+@media (max-width: 300px) {
+    .login-button {
+        width: 11em;
+        font-size: 1.4em;
+    }
+}
```
Comment (ParleJe) [inline src/Login.css:245]: is it possible to reduce number of breakpoints?

File: src/Login.css [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+.login-to-registration p {
+    font-size: 1.3em;
+    margin-top: 1em;
+    color: white;
+}
+
+.login-to-registration a {
+    font-size: 1.3em;
+    text-decoration: none;
+    color: #F06449;
```
Comment (ParleJe) [inline src/Login.css:260]: css variable

File: src/Login.css [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+}
+
+.login-to-registration {
+    display: contents;
+}
+
+.login-to-registration p {
+    font-size: 1.3em;
+    margin-top: 1em;
+    color: white;
```
Comment (ParleJe) [inline src/Login.css:254]: css variable

File: src/Register.css [hunk 12]
```diff
@@ -0,0 +1,141 @@
+.register {
+    margin: 0;
+    height: 100vh;
+    width: 100vw;
+    background: #1B1D2B;
```
Comment (ParleJe) [inline src/Register.css:5]: css variable

File: src/Register.css [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+    .register-about {
+        margin: auto 5em;
+    }
+}
+
+@media (max-width: 1600px) {
+    .register-about {
+        margin: auto 2em;
+    }
+}
```
Comment (ParleJe) [inline src/Register.css:40]: They override themselves

File: src/Register.css [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+    font-size: 1em;
+}
+
+.register-page input {
+    height: 1em;
+    width: 100%;
+    margin: 0.5em;
+    padding: 0.7em 0.5em;
+    border-style: none;
+    background: #292B2D;
```
Comment (ParleJe) [inline src/Register.css:105]: variable

File: src/Register.css [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+.register-icon {
+    margin-right: 0.3em;
+} 
+
+@media (max-width: 1600px) {
+    .register-page label div{
+        width: 30%;
+        font-size: 1em;
+    }
+}
```
Comment (ParleJe) [inline src/Register.css:121]: you have media query for 1600px above, use it

File: src/Register.css [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+    display: grid;
+    width: 12em;
+    height: 3em;
+    background: none;
+    border-style: solid;
+    justify-items: center;
+    align-items: center;
+    border-radius: 1.5em;
+    border-color: #F6F6F6;
+    color: #F6F6F6;
```
Comment (ParleJe) [inline src/Register.css:139]: css variables

File: src/Register.js [hunk 17]
```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import './Register.css';
+
+import {  FaLinkedin, FaGithub, FaFacebook } from "react-icons/fa";
```
Comment (ParleJe) [inline src/Register.js:5]: double spacebar on after curly bracket

File: src/Register.js [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+                        </div>  
+                    </form>
+                </div>
+            
+            </main>
+        </div>
+    );
+}
+
+export default Register;
```
Comment (ParleJe) [inline src/Register.js:76]: no new line on end of file

File: src/Login.js [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+                    <Link to='/register'>
+                        <a>Sing Up!</a>
+                    </Link>
+                </div>
+            </main>
+        </div>
+    );
+}
+
+export default Login;
```
Comment (ParleJe) [inline src/Login.js:66]: lack of new line at the end

File: src/Login.css [hunk 20]
```diff
@@ -0,0 +1,261 @@
...
+    color: #F6F6F6;
+}
+
+@media (max-width: 910px) {
+    .login-button {
+        width: 14.5em;
+        font-size: 1.4em;
+        margin: 0.4em 0;
+    }
+}
```
Comment (ParleJe) [inline src/Login.css:231]: you already have media query for 910px, merge them

File: src/Login.css [hunk 21]
```diff
@@ -0,0 +1,261 @@
...
+    background: #F6F6F6;
+    border-radius: 1em;
+    margin: 1.5em;
+}
+
+@media (max-width: 910px) {
+    .login-line {
+        display: none;
+    }
+}
```
Comment (ParleJe) [inline src/Login.css:210]: media query already exists for 910px

File: src/Login.css [hunk 22]
```diff
@@ -0,0 +1,261 @@
...
+      transform: rotate(0deg);
+    }
+    to {
+      transform: rotate(-360deg);
+    }
+}
+
+@media(max-width: 900px) {
+
+}
```
Comment (ParleJe) [inline src/Login.css:64]: empty query

File: src/Login.css [hunk 23]
```diff
@@ -0,0 +1,261 @@
...
+
+.login-header h1{
+    margin: 0;
+}
+
+@media (max-width: 910px) {
+    .login-header {
+        font-size: 1.5em;
+    }
+}
```
Comment (ParleJe) [inline src/Login.css:156]: same query above

File: src/Login.css [hunk 24]
```diff
@@ -0,0 +1,261 @@
...
+.login-header {
+    margin: 0;
+    font-style: italic;
+    font-size: 1.8em;
+    color: white;
+}
+
+.login-header h1{
+    margin: 0;
+}
```
Comment (ParleJe) [inline src/Login.css:150]: isn't it include in index css under "*"?

**Review by ParleJe — CHANGES_REQUESTED** [review]
Reposition media queries to the end of css, it's way more pleasant to view

**Review by ParleJe — CHANGES_REQUESTED** [review]
Formik to delete!!! It's not used dependency

File: package.json [hunk 25]
```diff
@@ -3,11 +3,18 @@
...
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^1.2.35",
+    "@fortawesome/free-brands-svg-icons": "^5.15.3",
+    "@fortawesome/free-solid-svg-icons": "^5.15.3",
+    "@fortawesome/react-fontawesome": "^0.1.14",
+    "@material-ui/core": "^4.11.3",
     "@testing-library/jest-dom": "^5.11.10",
     "@testing-library/react": "^11.2.6",
     "@testing-library/user-event": "^12.8.3",
+    "formik": "^2.2.6",
```
[the remark this answers is not in the document]
Comment (uniQueSense) [reply package.json]: Thank you for your attention. I hope this commit is the last one and you merge my branch.

**Review by ParleJe — APPROVED** [review]
OK

**Pull request merged**